### PR TITLE
server: introduce a safe mode for the server

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
DBAs often use a separate TiDB to perform some routine maintenance tasks. But due to some mistakes, often write SQL that has a great impact on the cluster.

### What is changed and how it works?
Introduce a safe mode for the server. `./tidb-server -safe-mode` will limit resource usage automatically.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
1. `./tidb-server -safe-mode`
1. Connect to TiDB
1. Check the variable e.g. `tidb-config`, and some concurrent variables.

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: tidb-server/main.go [hunk 1]
```diff
@@ -525,6 +531,26 @@ func overrideConfig() {
 	}
 }
 
+func setSafeModeConfig() {
+	if cfg.Store != "mocktikv" {
+		cfg.RunDDL = false
+	}
+	cfg.TokenLimit = 1
```
Comment (coocood) [inline tidb-server/main.go]: This is too small.

**Comment by coocood** [comment]
LGTM

File: tidb-server/main.go [hunk 2]
```diff
@@ -525,6 +531,15 @@ func overrideConfig() {
 	}
 }
 
+func setSafeModeConfig() {
+	if cfg.Store != "mocktikv" {
+		cfg.RunDDL = false
+	}
```
Comment (cfzjywxk) [inline tidb-server/main.go]: Is it needed to add some mem quota and token limits?

Comment (jackysp) [reply tidb-server/main.go]: @coocood suggests that it is no need to add limits for token. 😄

**Comment by cfzjywxk** [comment]
LGTM

**Comment by jackysp** [comment]
Closed after discussing with others. It is better to introduce a server-level scope for system variables to set some special variables for a special usage server, which will be started after 4.0 GA.

**Pull request closed without merging**